VCFv4.3.tex

```diff
@@ -564,23 +583,28 @@ \section{INFO keys used for structural variants}
 \end{verbatim}
 \normalsize
 \end{samepage}
-This key can be derived from the REF/ALT fields but is useful for filtering.
-The reserved values must be used for the types listed below:
+SVTYPE is a primitive basic data type and must be specified for all structural variants.
```

Comment:
Making SVTYPE mandatory raises another important question: what constitutes a structural variant? For example, a deletion can be any size from 1 bp to millions of bps. Does this mean that SVTYPE must now be specified for **all** deletions, including 1 bp? Or should a caller/user select a threshold for considering the variant “structural” and specifying SVTYPE in only that case? I think this needs to be clarified.

Reply:
From line 193 (the first paragraph after the list of ALT values):

 > Any symbolic alleles with a first level type other than one of those listed above are considered to be non-structural symbolic alleles.

The interpretation of that would be that a variant is structural if it uses a symbolic ALT allele from that list. For instance, a deletion of 1bp requires SVTYPE only if it uses ALT=<DEL>, not if it uses explicit alleles. I think that's ok.

Reply:
@jmmut Defining a structural variant based on whether it uses a symbolic allele is fine, but then it raises another problem: if SVTYPE is proposed to be used **when and only when** a structural variant symbolic allele is used, then SVTYPE appears to be completely redundant; unless it is expected that the type of a symbolic allele and the one in SVTYPE could be different.

Also, I've seen VCFs when alleles are specified with sequences but SVTYPE is still used. I'm not sure if we want to continue to allow that, but in any case, this needs to be described more explicitly

Reply:
It is redundant. I think tools rely on it for filtering and it's easier for humans to search the SVTYPE and the symbolic ALT, specially for BND where the ALT can be the notation with `[` and `]`. At least that's what I can extract from discussions in this repo.

Reply:
Right. In any case, redundancy of SVTYPE needs to be clarified in the spec. Should it always exactly mirror the symbolic allele? Or can we have combinations, like a symbolic allele of `<CNV>` and an SVTYPE of `DEL`? In that case, what combinations of symbolic alleles and SVTYPE are allowed?

Reply:
Has there been any further discussion about this? I'd like to propose a slight shift in the way things are defined which would mean that the current 'small' and 'structural' split can be removed/unified leading to a simpler and more robust API.

## Definitions

If we define 'variation' as 'a change relative to the reference sequence'. Then, in this context the term 'structural' can be further defined as 'where there is a change in length or arrangement relative to the reference sequence'. Under this definition SNV and MNV are not defined as structural. However, precisely defined, non-symbolic, insertions and deletions are defined a 'structural' regardless of their length. To fit with the existing spec there must be a distinction between 'implicit' and 'explicit' structural variation. Here 'implicit' would be a non-symbolic indel, precisely defined in the `REF` and `ALT` fields  and 'explicit' would include the extra fields used to define symbolic structural variants such as `SVTYPE`,`LEN/END` and `SVLEN`.

For example:

```tsv
#CHROM POS     ID        REF              ALT          QUAL FILTER INFO
1   1   .   A   .   .   PASS    # non-variation
1   1   .   A   T   .   PASS    # A SNV - this isn't a structural variation, but could have an implicit SVTYPE=SNV
1   1   .   A   TC  .   PASS    # an implicit `INS` here `SVTYPE` `LEN` and `SVLEN` can all be calculated from the REF and ALT alleles
1   1   .   AC  T   .   PASS    # an implicit `DEL` here `SVTYPE` `LEN` and `SVLEN` can all be calculated from the REF and ALT alleles
1   1   .   A  T   .   PASS    # an implicit `DEL` here `SVTYPE` `LEN` and `SVLEN` can all be calculated from the REF and ALT alleles
1   2827694 rs2376870   CGTGGATGCGGGGAC C   .   PASS    SVTYPE=DEL;LEN=15;SVLEN=-14 # an explicit, non-symbolic, precise deletion
2   321682  .   T   <DEL>   .   PASS    SVTYPE=DEL;LEN=206;SVLEN=-205;CIPOS=-56,20;CIEND=-10,62 # explicit, symbolic, imprecise deletion
```

So the only real difference here is that of the implicit and explicit declaration of the `SVTYPE`, `LEN` and `SVLEN`. Explicit types _must_ be used with symbolic alleles and *can* be used with non-symbolic alleles. 

All the other symbolic alleles follow the existing specifications. In order to maintain consistency between symbolic and non-symbolic types using the `SVTYPE` key the values `SNV` and `MNV` would need adding to the reserved set. Strictly `SVTYPE` ought to be renamed `TYPE` to be able to cover all the types of variation so applications can treat variation in a uniform manner, but its workable without doing this.

The upshot of this is a clean and consistent API in things like the HTSJDK where a client needs to call `VariantContext.getType()` and `getStructuralType()` depending on the return values. For example `INDEL` or `MIXED` in the case of `getType()` not being `SNV`/`MNV` or `null` being returned from `getStructuralType()` when the ALT allele isn't symbolic. Furthermore it is more in line with the way variation is reported in the HGVS, at least in the way they categorise some variation e.g. [Substitution](http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/), [Deletion](http://varnomen.hgvs.org/recommendations/DNA/variant/deletion/), [Insertion](http://varnomen.hgvs.org/recommendations/DNA/variant/insertion/), [Inversion](http://varnomen.hgvs.org/recommendations/DNA/variant/inversion/), which marginally helps applications ingesting VCF and reporting findings in HGVS.

Does this sound workable or sheer lunacy?

Reply:
@julesjacobsen Thank you for your comment! Since we're trying to merge this PR with minimal changes, I've separated it into a separate issue: https://github.com/samtools/hts-specs/issues/547